### PR TITLE
Optionals for additem and quick bug fix

### LINKED ISSUE #19: Make quantity, description and location inputs for items optional, set default values

### LINKED ISSUE #77: Bug Report - ID has duplicate values
image

assume all items from 1...n exist delete item of id n-1 in the list
then the num items in list becomes n-1
exit application
item counter++ = n-1
then addi item1 -> counter = n, 
BUG: 2 items with id n

### PR DESCRIPTION
**Pull request by blackonyyx**
resolves #77 
resolves #19 
Reposting commit message for ease of reference:

Add tagging to items, enhanced location parsing.
    
ID will now monotonically increase for both recipes and item ids.
Resolve Duplicate ID bug.
Slight "issue" which is improbable in real usecase,
but theoretically possible is that an user inputs Integer.MAX_VALUE - 1 times
addi -n asdf 
This is because first instance will create the item, but then second onwards will
increase id counter without adding any item, then add a new item.

addi -n IKillInventory
Then, exit Inventoryinator
Then reopen Inventoryinator
addi -n NowDie
will cause integer overflow, and ID will be negative, causing an error in ID Assertion error.

Analysis:
It is improbable that any real user will perform such a perversed usecase, hence this edge case error scenario is acceptable.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -95,17 +95,19 @@ Will eventually be extended to get help about specific commands.
...
 	 
-- `addi` **-n** \<**item name**\> \[-q \<qty\>\] \[-d \<desc\>\] \[-l \<location1, location2…\>\] 
+- `addi` **-n** \<**item name**\> \[-q \<qty\>\] \[-d \<desc\>\] \[-l \<location1, location2…\>\] \[-t \<tag1, tag2\>\]
 
 **DESCRIPTION:**
 
 - **q:**	qty indicates quantity to add (default: 1)
 - **d:**	desc indicates description of item (default: “No description given.”)
 - **l:**	locations indicate where item is found 
+- **t:**    user defined **tag** for an item
```
Comment (aqcd) [inline docs/UserGuide.md]: space equally as the above lines

File: src/main/java/seedu/address/logic/commands/EditItemCommand.java [hunk 2]
```diff
@@ -35,9 +36,11 @@
             + "[" + PREFIX_ORIGINAL_ITEM_NAME + "NAME] "
             + "[" + PREFIX_ITEM_NAME + "NAME] "
             + "[" + PREFIX_ITEM_QUANTITY + "QUANTITY] "
-            + "[" + PREFIX_ITEM_DESCRIPTION + "DESCRIPTION] \n"
+            + "[" + PREFIX_ITEM_DESCRIPTION + "DESCRIPTION] "
+            + "[" + PREFIX_ITEM_TAG + "TAG/TAGS] "
             + "Example: " + COMMAND_WORD + " 1 "
-            + PREFIX_ITEM_QUANTITY + "20 ";
+            + PREFIX_ITEM_QUANTITY + "20 "
+            + PREFIX_ITEM_TAG + "tag1, tag2,tag3 , tag4";
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/commands/EditItemCommand.java:43]: spacing after `,`

File: src/main/java/seedu/address/logic/parser/EditItemCommandParser.java [hunk 3]
```diff
@@ -22,10 +26,16 @@
      */
     public EditItemCommand parse(String args) throws ParseException {
         requireNonNull(args);
+
+
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/parser/EditItemCommandParser.java]: remove extra line

File: src/test/java/seedu/address/logic/parser/EditItemCommandParserTest.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package seedu.address.logic.parser;
```
Comment (aqcd) [inline src/test/java/seedu/address/logic/parser/EditItemCommandParserTest.java:1]: add test to edit tags

File: src/main/java/seedu/address/model/location/UniqueLocationList.java [hunk 5]
```diff
@@ -48,6 +49,19 @@ public int findLocationID(Location toFind) {
...
+     * @return an Optional Location that maybe exists.
+     */
+    public Optional<Location> findLocationFromId(int id) {
+        requireNonNull(id);
+        return internalList.stream()
+                .filter(x -> x.getId() == id)
+                .findFirst();
+    }
+
+
```
Comment (aqcd) [inline src/main/java/seedu/address/model/location/UniqueLocationList.java]: add test for this

**Review by aqcd — CHANGES_REQUESTED** [review]
Needs tests for tags in item and precursor, and perhaps add item command parser.

File: src/main/java/seedu/address/logic/commands/EditItemCommand.java (hunk 2 above)
Comment (blackonyyx) [reply src/main/java/seedu/address/logic/commands/EditItemCommand.java:43]: this is intentional to show user that comma spacing is irrelevent.

File: docs/diagrams/sequenceDiagrams/DeleteItemSequenceDiagram.puml [hunk 6]
```diff
@@ -3,67 +3,70 @@
 
 box Logic LOGIC_COLOR_T1
 participant ":LogicManager" as LogicManager LOGIC_COLOR
-participant ":InventoryParser" as InventoryParser LOGIC_COLOR
-participant ":DeleteCommandParser" as DeleteCommandParser LOGIC_COLOR
-participant "d:DeleteCommand" as DeleteCommand LOGIC_COLOR
+participant ":InventoryinatorParser" as InventoryinatorParser LOGIC_COLOR
+participant ":DeleteItemCommandParser" as DeleteCommandParser LOGIC_COLOR
+participant "d:DeleteItemCommand" as DeleteCommand LOGIC_COLOR
```
Comment (aqcd) [inline docs/diagrams/sequenceDiagrams/DeleteItemSequenceDiagram.puml]: standardise to without the "inator", same for rest of instances

**Pull request merged**